servicetalk-grpc-netty/src/test/java/io/servicetalk/grpc/netty/ProtocolCompatibilityTest.java

```diff
@@ -165,295 +174,369 @@ public SocketAddress listenAddress() {
+                                                        @FromDataPoints("streaming") final boolean streaming,
+                                                        @FromDataPoints("compression") final String compression)
             throws Exception {
-        final TestServerContext server = serviceTalkServer(ErrorMode.STATUS, ssl);
-        final CompatClient client = serviceTalkClient(server.listenAddress(), ssl);
+        final TestServerContext server = serviceTalkServer(ErrorMode.STATUS, ssl, compression);
+        final CompatClient client = serviceTalkClient(server.listenAddress(), ssl, compression);
         testGrpcError(client, server, true, streaming);
     }
 
     @Theory
     public void serviceTalkToServiceTalkErrorWithStatusViaServiceFilter(
-            @FromDataPoints("ssl") final boolean ssl, @FromDataPoints("streaming") final boolean streaming)
+            @FromDataPoints("ssl") final boolean ssl, @FromDataPoints("streaming") final boolean streaming,
+            @FromDataPoints("compression") final String compression)
             throws Exception {
-        final TestServerContext server = serviceTalkServer(ErrorMode.STATUS_IN_SERVICE_FILTER, ssl);
-        final CompatClient client = serviceTalkClient(server.listenAddress(), ssl);
+        final TestServerContext server = serviceTalkServer(ErrorMode.STATUS_IN_SERVICE_FILTER, ssl, compression);
+        final CompatClient client = serviceTalkClient(server.listenAddress(), ssl, compression);
         testGrpcError(client, server, true, streaming);
     }
 
     @Theory
     public void serviceTalkToServiceTalkErrorWithStatusViaServerFilter(
-            @FromDataPoints("ssl") final boolean ssl, @FromDataPoints("streaming") final boolean streaming)
+            @FromDataPoints("ssl") final boolean ssl, @FromDataPoints("streaming") final boolean streaming,
+            @FromDataPoints("compression") final String compression)
             throws Exception {
-        final TestServerContext server = serviceTalkServer(ErrorMode.STATUS_IN_SERVER_FILTER, ssl);
-        final CompatClient client = serviceTalkClient(server.listenAddress(), ssl);
+        final TestServerContext server = serviceTalkServer(ErrorMode.STATUS_IN_SERVER_FILTER, ssl, compression);
+        final CompatClient client = serviceTalkClient(server.listenAddress(), ssl, compression);
         testGrpcError(client, server, true, streaming);
     }
 
     private static void testBlockingRequestResponse(final BlockingCompatClient client, final TestServerContext server,
-                                                    final boolean streaming) throws Exception {
+                                                    final boolean streaming,
+                                                    @Nullable final String compresion) throws Exception {
         try {
+            final ContentCodec codec = serviceTalkCodingFor(compresion);
```

Comment:
The `serviceTalkCodingFor` method returns `identity()` when `compresion == null`. Can we return `null` in this case and use `ScalarCallMetadata.INSTANCE` later? It will verify a case when compression is not specified, similar to `client.scalarCall(request)` (without meta-data param).
Same for the streaming variant, we can use `ClientStreamingCallMetadata.INSTANCE`.